chia/data_layer/data_store.py

```diff
@@ -376,16 +376,17 @@ async def get_roots_between(
         return roots
 
     async def get_last_tree_root_by_hash(
-        self, tree_id: bytes32, hash: bytes32, max_generation: Optional[int] = None, *, lock: bool = True
+        self, tree_id: bytes32, hash: Optional[bytes32], max_generation: Optional[int] = None, *, lock: bool = True
     ) -> Optional[Root]:
         async with self.db_wrapper.locked_transaction(lock=lock):
             max_generation_str = f"AND generation < {max_generation} " if max_generation is not None else ""
+            node_hash_str = "AND node_hash == :node_hash " if hash is not None else "AND node_hash is NULL "
```

Comment:
I wasn't aware that you can't equality check to `NULL`.  That's a pain.

Reply:
not sure about SQLite, but for SQL this generally seems to cause problems, as `NULL == NULL` returns false:

https://stackoverflow.com/questions/9581745/sql-is-null-and-null

I think it's safer to keep it in this form regardless.

Reply:
Yeah, I was going to suggest a change but went and tested it first.  I agree it needs to be this way.